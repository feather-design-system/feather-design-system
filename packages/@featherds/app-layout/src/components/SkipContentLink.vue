<template>
  <a class="skip" :href="contentId" @click="scrollTo">{{ skipLabel }}</a>
</template>

<script lang="ts">
import { defineComponent } from "vue";
export const props = {
  content: {
    type: String,
    required: true,
  },
  skipLabel: {
    type: String,
    required: true,
  },
} as const;
export default defineComponent({
  props,
  computed: {
    contentId() {
      return `#${this.content}`;
    },
  },
  methods: {
    scrollTo(e: MouseEvent) {
      if (!Object.hasOwn(this, "$router")) {
        return;
      } else {
        e.preventDefault();
        const scrollEl = document.getElementById(this.content);
        if (scrollEl) {
          scrollEl.scrollIntoView({ behavior: "smooth" });
          scrollEl.tabIndex = -1;
          scrollEl.focus();
        }
      }
    },
  },
});
</script>

<style lang="scss" scoped>
@import "@featherds/styles/mixins/elevation";
@import "@featherds/styles/mixins/typography";
a.skip {
  width: 1px;
  height: 1px;
  overflow: hidden;
  @include elevation(2);
  @include button();

  background-color: var($surface);
  padding: 0.25rem;
  position: absolute;
  top: 0px;
  left: 0px;
  border: 1px solid var($primary);
  &:focus {
    width: auto;
    height: auto;
    z-index: var($zindex-tooltip);
  }
}
</style>
