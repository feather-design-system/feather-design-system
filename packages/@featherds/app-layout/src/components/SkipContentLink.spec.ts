import { mount } from "@vue/test-utils";
import SkipContentLink from "./SkipContentLink.vue";
let scrollIntoViewMock = jest.fn();
window.HTMLElement.prototype.scrollIntoView = scrollIntoViewMock;

const getWrapper = () =>
  mount(SkipContentLink, {
    props: {
      content: "test",
      skipLabel: "This is a text passage",
    },
    attachTo: "body",
  });

beforeEach(() => {
  document.body.innerHTML = `
    <div id="test"></div>
  `;
});
describe("SkipContentLink", () => {
  it("should match snapshot", () => {
    const wrapper = getWrapper();
    expect(wrapper.element).toMatchSnapshot();
  });
  it("should use smooth scroll", () => {
    const wrapper = getWrapper();
    wrapper.trigger("click");
    expect(scrollIntoViewMock).toHaveBeenCalled();
    expect(document.activeElement?.id).toBe("test");
  });
});
