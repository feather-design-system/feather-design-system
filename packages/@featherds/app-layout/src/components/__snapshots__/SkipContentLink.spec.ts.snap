// Vitest Snapshot v1

exports[`SkipContentLink > should match snapshot 1`] = `
<a
  class="skip"
  href="#test"
>
  This is a text passage
</a>
`;
